develop/blockchains/testing/setup.md

```diff
@@ -1,4 +1,54 @@
+
+In the context of Polkadot SDK, runtime testing involves creating a simulated environment that mimics real blockchain conditions. This type of testing goes beyond individual pallet validation, focusing on how multiple components integrate and collaborate across the system.
+
+While unit tests provide confidence that individual pallets function correctly in isolation, runtime tests offer a holistic view. These tests validate pallets’ communication and interaction, ensuring a seamless and functional blockchain system. By running integration tests at the runtime level, you can catch issues that only arise when multiple pallets are combined, which is critical for building a stable and reliable blockchain.
+
+### Mocking the Runtime
+
+To thoroughly test a runtime, you'll need a mock environment that simulates the behavior of a live blockchain. This involves creating a module that replicates runtime execution, allowing you to evaluate how the system behaves before deploying to a live network.
+
+The mock runtime includes all the necessary pallets and configurations needed for testing. To simplify the process, you can create a module that integrates all components, making it easier to assess how pallets and system elements interact.
+
+Here’s a simple example of how to create a testing module that simulates these interactions:
+
+```rust
+--8<-- 'code/develop/blockchains/testing/setup/integration-testing-module.rs'
+```
+
+!!! note
+    This snippet `crate::*;`, which imports all the necessary components from your crate (including runtime configurations, pallet modules, and utility functions) into the `integration_testing` module. This allows you to write tests without manually importing each piece, making the code more concise and readable.
+
+Once the testing module is set, the next step is configuring the genesis storage—the initial state of your blockchain. Genesis storage sets the starting conditions for the runtime, defining how pallets are configured before any blocks are produced.
+
+In Polkadot SDK, we can create this storage using the [`BuildStorage`](https://paritytech.github.io/polkadot-sdk/master/sp_runtime/trait.BuildStorage.html){target=\_blank} trait from the [`sp_runtime`](https://paritytech.github.io/polkadot-sdk/master/sp_runtime){target=\_blank} crate. This trait is essential for building the configuration that initializes the blockchain's state.
+
+The function `new_test_ext()` demonstrates setting up this environment. It uses `frame_system::GenesisConfig::<Runtime>::default()` to generate a default genesis configuration for the runtime, followed by `.build_storage()` to create the initial storage state. This storage is then converted into a format usable by the testing framework, [`sp_io::TestExternalities`](https://paritytech.github.io/polkadot-sdk/master/sp_io/type.TestExternalities.html){target=\_blank}, allowing tests to be executed in a simulated blockchain environment.
+
+Here’s the code that sets up the mock runtime:
```

Comment:
It would be cool to extend a little bit more and show how to add some things to the genesis state, such as balances